docs/websockets/ewallet_api.md

```diff
@@ -0,0 +1,499 @@
+
+For authentication from servers:
+```
+Authorization: OMGServer base64(access_key:secret_key)
+```
+
+From clients:
+```
+Authorization: OMGClient base64(api_key:authentication_token)
+```
+
+The `Accept` header also needs to be set to the appropriate media type (replace `v1` with the version you want to use):
+
+```
+Accept: application/vnd.omisego.v1+json
+```
+
+Not sending valid headers will result in the request failing and the server returning some kind of invalid HTTP upgrade error.
+
+## Joining a channel
+
+Channels can be joined by sending the `phx_join` event. The list of events is available below. Joining allows you to listen to its events and, later on, send requests to retrieve data.
+
+## Listening for events
+
+Events for a specified chahnel will be sent to everyone who joined the channel. See list of channels and potential events below.
+
+## Leaving a channel
+
+To leave a channel, simply send the `phx_leave` event.
+
+Payload sent:
+```
+{
+  "topic": "transaction_request:some_id",
+  "event": "phx_leave",
+  "ref": "1",
+  "data": {}
+}
+```
+
+## Keeping the connection alive (hearbeat)
+
+In order to keep the socket connection alive, you can periodically send heartbeats using the following payload.
+
+Payload sent:
+```
+{
+  "topic": "phoenix",
+  "event": "heartbeat",
+  "ref": "1",
+  "data": {}
+}
+```
+
+## Channels
+
+The following channels are available in the eWallet API:
+
+### `account:{id}`
+
+All events related to the specified account will be sent to that channel.
+
+Potential events:
+
+- `transaction_consumption_request`
+- `transaction_consumption_finalized`
+
+### `user:{id}`
+
+All events related to the specified user will be sent to that channel.
+
+Potential events:
+
+- `transaction_consumption_request`
+- `transaction_consumption_finalized`
+
+### `address:{address}`
+
+All events related to the specified address will be sent to that channel.
+
+Potential events:
+
+- `transaction_consumption_request`
+- `transaction_consumption_finalized`
+
+### `transaction_request:{id}`
+
+All events related to the specified transaction request will be sent to that channel.
+
+Potential events:
+
+- `transaction_consumption_request`
+- `transaction_consumption_finalized`
+
+### `transaction_consumption:{id}`
+
+All events related to the specified consumption will be sent to that channel.
+
+Potential events:
+
+- `transaction_consumption_finalized`
+
+## Events
+
+### Sendable events
+
+- `phx_join`: event used to join a channel.
+
+```
+{
+  "topic": "transaction_request:some_id",
+  "event": "phx_join",
+  "ref": "1",
+  "data": {}
+}
+```
+
+- `phx_leave`: event used to leave a channel.
+
+```
+{
+  "topic": "transaction_request:some_id",
+  "event": "phx_leave",
+  "ref": "2",
+  "data": {}
+}
+```
+
+- `heartbeat`: event used to keep the connection open.
+
+```
+{
+  "topic": "phoenix",
+  "event": "heartbeat",
+  "ref": "1",
+  "data": {}
+}
+```
+
+### Receivable system events
```

Comment:
Not sure if these `phx_` event prefix will confuse users since they don't need to know that it's run by Phoenix. Maybe it could be named something else?

Doesn't need to be in this PR but want to bring this up for discussion.

Reply:
Could it be `omg_` instead? 😃

Reply:
I wish, but it's too much work to change... it's really embedded inside Phoenix. :/

Reply:
Thought so. Shouldn't be in this PR but would be nice to have in the future I guess.